### PR TITLE
FOGL-6670 Developer Features and OMF Types in troubleshooting OMF

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: docs/troubleshooting_pi-server_integration.rst [hunk 1]
```diff
@@ -118,6 +122,126 @@ Some error messages and causes:
     * - North_Readings_to_PI[24485]: ERROR: Sending JSON data error : **Container not found**. 4273005507977094880_1measurement_sin_4816_asset_1 - WIN-4M7ODKB0RH2:443 /piwebapi/omf
       - Fledge is able to interact with PI Web API but there is an attempt to store data in a PI Point that does not exist.
 
+OMF Plugin Data
+===============
+
+The OMF north plugin must create type information within the OMF subsystem of the PI server before any data can be sent. This type information is persisted within the PI Server between sessions and must also be persisted within Fledge for each connection to a PI Server. This is done using the plugin data persistence features of the Fledge north plugin.
```
Comment (RVerhoeff) [inline docs/troubleshooting_pi-server_integration.rst:128]: PI Server

File: docs/troubleshooting_pi-server_integration.rst [hunk 2]
```diff
@@ -118,6 +122,126 @@ Some error messages and causes:
...
+
+        - Remove the type data from the PI Server such that neither end has the type information.
+
+        - Before deleting the north service or task export the plugin persisted data and import that data into the new service or task.
+
+Cleanup a PI server and reuse and existing OMF North service or task
+    This is the opposite problem to that stated above, the plugin will try to send data thinking that the types have already been created in the PI Server and receive an error. Fledge will automatically correct for this and create new types. These new types however will be created with new names, which may not be the desired behavior. Type names are created using a fixed algorithm. To re-use the previous names stopping the north service and deleting the plugin persisted data will reset the algorithm and recreate the types using the names that had been previously used.
```
Comment (RVerhoeff) [inline docs/troubleshooting_pi-server_integration.rst:140]: Add comma: "To re-use the previous names, stopping..."

File: docs/troubleshooting_pi-server_integration.rst [hunk 3]
```diff
@@ -118,6 +122,126 @@ Some error messages and causes:
...
+
+Taking an existing Fledge north task or service and moving it to a new PI Server
+    This new PI server will not have the type information from the old and we will once again get errors when sending data due to these missing types. Fledge will automatically correct for this and create new types. These new types however will be created with new names, which ma y not be the desired behavior. Type names are created using a fixed algorithm. To re-use the previous
```
Comment (RVerhoeff) [inline docs/troubleshooting_pi-server_integration.rst:143]: Space in word: "which ma y not be the desired behavior"

File: docs/troubleshooting_pi-server_integration.rst [hunk 4]
```diff
@@ -118,6 +122,126 @@ Some error messages and causes:
...
+names stopping the north service and deleting the plugin persisted data will reset the algorithm and recreate the types using the names that had been previously used.
```
Comment (RVerhoeff) [inline docs/troubleshooting_pi-server_integration.rst:144]: Add comma: "To re-use the previous names, stopping..."

File: docs/troubleshooting_pi-server_integration.rst [hunk 5]
```diff
@@ -118,6 +122,126 @@ Some error messages and causes:
...
+
+Managing Plugin Persisted Data
+------------------------------
+
+This is not a feature that users would ordinarily need to be concerned with, however it is possible to enable *Developer Features* in the Fledge User Interface that will provide a mechanism to manage this data.
+
+Enable Develop Features
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Navigate to the *Settings* page of the GUI and toggle on the *Developer Features* tick box on the bottom left of the page.
```
Comment (RVerhoeff) [inline docs/troubleshooting_pi-server_integration.rst:154]: _check_ box?

File: docs/troubleshooting_pi-server_integration.rst [hunk 6]
```diff
@@ -118,6 +122,126 @@ Some error messages and causes:
...
++-----------------+-------------------------------------------------------------------------------------------+
+| hintChecksum    | A checksum of the OMFHints used to create this type. 0 if no OMF Hint was used.           |
++-----------------+-------------------------------------------------------------------------------------------+
+| namingScheme    | The current OMF naming scheme when the type was sent.                                     |
++-----------------+-------------------------------------------------------------------------------------------+
+| afhHash         | A Hash of the AF settings for the type.                                                   |
++-----------------+-------------------------------------------------------------------------------------------+
+| afHierarchy     | The AF Hierarchy location.                                                                |
++-----------------+-------------------------------------------------------------------------------------------+
+| afHierarchyOrig | The original setting of AF Heirarchy. This may differ from the above if specific AF rules |
```
Comment (RVerhoeff) [inline docs/troubleshooting_pi-server_integration.rst:238]: Should be "Hierarchy"

**Review by RVerhoeff — CHANGES_REQUESTED** [review]
I commented on using "PI Server" instead of "PI server" in one place but it applies throughout.

**Review by ashwinscale — APPROVED** [review]
LGTM

**Review by RVerhoeff — APPROVED** [review]

**Review by ashwinscale — APPROVED** [review]
LGTM

**Pull request merged**